docs/md_merge.py

```diff
@@ -0,0 +1,357 @@
+                if not linked_path.exists():
+                    raise ValueError(f"Linked file '{str(linked_path)}' does not exist! (link in '{str(file_path)}')")
+                if file_path.samefile(linked_path):
+                    for link_name in link_names:
+                        link_anchor = f'<a name="{link_name}"></a>'
+                        output_file.write(link_anchor + "\n")
+
+            for line in content:
+                text = line
+                if isinstance(line, HeaderLine):
+                    text = line.to_text(new_depth=depth)
+                if isinstance(line, LinkLine):
+                    text = line.to_text()
+                output_file.write(text + "\n")
+            output_file.write("\n")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Merge multiple markdown files into one."
+    )
+    parser.add_argument(
+        "-i",
+        "--input",
+        dest="input",
+        default=".",
+        type=handle_input_path,
+        help="path to the input folder",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        dest="output",
+        default="./out/",
+        type=handle_output_path,
+        help="path to the output folder",
+    )
+    parser.add_argument(
+        "-n",
+        "--name",
+        dest="name",
+        default="output.md",
+        type=str,
+        help="name of the output document",
+    )
+    parser.add_argument(
+        "-l",
+        "--lang",
+        dest="language",
+        default="en",
+        type=str,
+        help="the language code used in the filenames (.<language>.md)",
+    )
+    parser.add_argument(
+        "-d",
+        "--depth",
+        dest="depth",
+        default=1,
+        type=int,
+        help="the initial depth of headers to consider, 1 will prepend a single '#' to header lines,"
+             " another '#' will be added whenever descending into a subdirectory"
+    )
+    parser.add_argument(
+        "-m",
+        "--max-depth",
+        dest="max_depth",
+        default=-1,
+        type=int,
+        help="the maximum depth for stepping into subdirectories, -1 disables this feature"
+    )
+    parser.add_argument(
+        "--paths-relative-to-input-dir",
+        dest="paths_relative_to_input_dir",
+        default=False,
+        action=BooleanOptionalAction,
+        type=bool,
+        help="when this is set, paths inside markdown files are interpreted as relative to the top level input"
+             " directory, otherwise, they are treated as relative to the markdown files they are contained in (default)"
+    )
+    return parser.parse_args()
+
+
+def handle_input_path(path):
+    if os.path.isdir(path):
+        return path
+    else:
+        raise argparse.ArgumentTypeError(f"'{path}' is not a valid path")
+
+
+def handle_output_path(path):
+    if os.path.isdir(path):
+        return path
+    else:
```

Comment:
Nit: You may drop the noisy else and shift-left the current else block.

Reply:
updated